### PR TITLE
get rid of the potential file inclusion via variable

### PR DESCRIPTION
**Pull request by yanghaichao12**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
get rid of the potential file inclusion via variable

Which issue(s) this PR fixes (optional, in fixes #(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes #

Special notes for your reviewer:

Does this PR introduce a user-facing change?:

NONE

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/ok-to-test
/priority backlog

@yanghaichao12 
could you please provide more details (examples) of what the problem is here?
this seems to me like more of a bug than a cleanup.
thanks.

File: pkg/proxy/ipvs/proxier.go [hunk 1]
```diff
@@ -480,7 +481,7 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 		return nil, err
 	}
 	builtinModsFilePath := fmt.Sprintf("/lib/modules/%s/modules.builtin", kernelVersion)
-	b, err := ioutil.ReadFile(builtinModsFilePath)
+	b, err := ioutil.ReadFile(filepath.Clean(builtinModsFilePath))
```
Comment (liggitt) [inline pkg/proxy/ipvs/proxier.go]: shouldn't this run on kernelVersion before we include it in a filepath, and disallow a cleaned kernelVersion that contains multiple path segments?

**Comment by yanghaichao12** [comment]
> /ok-to-test
> /priority backlog
> 
> @yanghaichao12
> could you please provide more details (examples) of what the problem is here?
> this seems to me like more of a bug than a cleanup.
> thanks.

**Comment by yanghaichao12** [comment]
@neolit123 thanks  for your reply. sorry for the slow response.  
I am just worried that the function "GetKernelVersionAndIPVSMods" will not output a cleaned path with some errors made by someone. 
as for the value of "/proc/sys/kernel/osrelease", it seems unlikely to change unless the kernel is rebuilded(maybe there's other way,i'm not very sure. If this is the case, it may be really dangerous.）
thanks  for your reply，again.

File: pkg/proxy/ipvs/proxier.go (hunk 1 above)
Comment (yanghaichao12) [reply pkg/proxy/ipvs/proxier.go]: hi, thanks for your replay. As  a variable, "kernelVersion" maybe begin with "/" or something else.  so i think if i run on kernelVersion firstly,maybe the "builtinModsFilePath" is still not cleaned.
in fact, i did it at the beginning，but still get warning from GoSec.... :(

**Comment by liggitt** [comment]
cleaning the entire path once it's been constructed doesn't really accomplish much. if you want to avoid path traversal via a kernelVersion value like "../../../../../foo", you have to add things to clean or detect multiple path segments or backsteps in kernelVersion before including it in the path

/close

**Comment by yanghaichao12** [comment]
@liggitt ok :)

**Comment by yanghaichao12** [comment]
@liggitt hi，how about like this： 'builtinModsFilePath := filepath.Join("/lib/modules/",filepath.Clean(kernelVersion),"/modules.builtin")'？

**Pull request closed without merging**